### ISSUE TITLE
WO-023: input holding loss, targeted audits, soft budget

### ISSUE THREAD

**Issue by Xocas12** [Xocas12/gosplan-env#28]
> ## NOT ISSUABLE YET - THIS CARD IS A PLACEHOLDER AND ITS CONTENT IS PROVISIONAL
>
> **This work order is issued only AFTER the Phase-2 spec revision.** It is gated on gate **G2** and
> on the lead's spec revision that follows G2. Nothing here may be started before both have happened.
>
> **The behaviour of all three mechanisms below is frozen at that revision, and not before.** PLAN
> §12.4 gives WO-023 as one summary line; PLAN §2.7.4 gives the targeted-audit rule with a
> coefficient (`κ_t`) that is **not yet a configuration field**; PLAN §3 gives the soft budget only
> as "P2 definition: bailout probability on `fill < 1`" - a definition of the parameter, not of how
> the bailout acts.
>
> **Everything under "Implementation notes" below is PROVISIONAL** until the lead revises the spec.
> Do not read it as settled and do not present it to anyone else as settled. The soft-budget item in
> particular is under-specified in a way that touches CONTRACT rule 4.
>
> **An implementer who reaches the soft budget before the revision files an AMBIGUITY REPORT and
> stops** (`workorders/AMBIGUITY_TEMPLATE.md`, CONTRACT rule 3).

> **WO-023** - Phase 2 - Owner **MID-strong** - Difficulty **set by the LEAD at issue** (PLAN §12.4 fixes the tier only) - Gate **G3**
> Card: [`workorders/WO-023.md`](https://github.com/Xocas12/gosplan-env/blob/main/workorders/WO-023.md)

## What this is

Three separate Phase-2 mechanisms that share a card because they all bear on hoarding. (1) An
**input holding loss** `h_X`: inputs sitting in stock decay, so hoarding them is not free - in Phase
1 the hoarding cost is 0 by design, which makes hoarding trivially optimal and therefore
uninformative. (2) **Targeted audits**: instead of auditing at a flat rate, the planner audits an
enterprise more often when it can see shortfall downstream of it, so misreporting has a
state-dependent cost. (3) A **soft budget**: a probability of a bailout when an enterprise's input
fill falls short, which is the classic reason a Soviet enterprise over-requests inputs. Only the
mechanisms are here; the hoarding phenomenon itself (PLAN §4.1 row 5) is held out and is measured
for the first time in WO-030.

## Depends on

- gate G2 - the Phase-1 gate (PLAN §4.5 criteria 1-4, signed off Human + LEAD). This card cannot be
  started before G2 has passed.
- the **P2 spec revision** (LEAD) - which must, among other things, add `κ_t` to the spec and pick
  the soft-budget mechanism.
- WO-013
- WO-004
- WO-006
- WO-007
- WO-009

Consumed by WO-030 (phenomenon row 5) and WO-031 (the P2 acceptance run).

**Blocked by:** #18 (WO-013), #9 (WO-004), #11 (WO-006), #12 (WO-007), #14 (WO-009), #45 (gate G2)
This card may not be started until every issue above is closed (CONTRACT rule 12 / the loop in CONTRIBUTING.md).

## Read first (exhaustive whitelist)

- `PLAN.md` §§ 2.7.3, 2.7.4, 2.8, 2.9.1, 2.11, 2.15, 3, 4.2, 4.3 - PLAN.md is git-ignored and
  supplied out of band; it is not in the repository and has no blob link.
- [`CONTRACT.md`](https://github.com/Xocas12/gosplan-env/blob/main/CONTRACT.md) - rules 4, 5, 7, 9, 11
- [`spec/spec.py`](https://github.com/Xocas12/gosplan-env/blob/main/spec/spec.py) -
  `InformationConfig.audit_mode`, `.shortfall_visibility`, `.audit_noise`,
  `IncentiveConfig.soft_budget`, `SupplyConfig.input_holding_loss`,
  `PlannerView.downstream_shortfall`, `select_audits`, `deliver`, `enterprise_reward`
- [`spec/CHANGELOG.md`](https://github.com/Xocas12/gosplan-env/blob/main/spec/CHANGELOG.md)
- [`gosplan/config.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/config.py)
- [`gosplan/params.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/params.py)
- [`gosplan/rng.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/rng.py)
- [`gosplan/env/planner.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/planner.py)
- [`gosplan/env/reporting.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/reporting.py)
- [`gosplan/env/reward.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/reward.py)
- [`gosplan/env/state.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/state.py)
- [`docs/params_sources.md`](https://github.com/Xocas12/gosplan-env/blob/main/docs/params_sources.md)
- **the revised spec text, which does not exist yet - that is what gates this card**

Read nothing else. CONTRACT rule 12.

## Write only

Provisional; the revision fixes the final list.

- `gosplan/env/planner.py` - the `targeted` branch of audit selection; `downstream_shortfall` into
  the planner view; the soft-budget hook at DELIVER, if that is where the revision puts it
- `gosplan/env/state.py` - the per-period input holding loss on `inv_inputs`
- `gosplan/env/reporting.py` - only if the revision routes the bailout through the penalty

Not this card: `ref/`, `tests/**`, `spec/spec.py` (adding a `κ_t` field is a LEAD change with a
`spec/CHANGELOG.md` entry and an arm classification - CONTRACT rules 1, 11),
`gosplan/metrics/phenomena.py`.

Creating, editing or deleting any other file is a CONTRACT rule 12 violation.

## Must pass

**No new test ids are frozen yet** - the revision writes them and names their ids in this section
(CONTRACT rule 12).

Already binding and already determinate, green after this change:

- [ ] **T-U1** conservation to 1e-9, with the `h_X` loss on input stocks as an explicit term -
      `tests/unit/test_conservation.py`
- [ ] **T-U8** penalty: `positive_part` gives 0 for any under-report; `audited = False` gives 0 -
      `tests/unit/test_reporting.py`
- [ ] **T-B4** planner blindness - `downstream_shortfall` reaches the planner only through
      `make_planner_view`, and only as the noisy, visibility-gated quantity -
      `tests/behavioural/test_planner_blindness.py`
- [ ] **T-B7** golden parity with `ref/` - `tests/golden/test_golden_parity.py`

Acceptance criterion PLAN already fixes: **at the Phase-1 defaults (`input_holding_loss = 0.0`,
`audit_mode = "random"`, `shortfall_visibility = 0.0`, `soft_budget = 0.0`) all three mechanisms are
exactly inert**, so every golden file passes unchanged. PLAN §2.7.4 additionally makes `targeted`
conditional on `shortfall_visibility > 0`: with no visibility there is nothing to target on, and the
branch must reduce to `random`.

Completion command, copy-paste:

```bash
uv run ruff check . && uv run ruff format --check . && uv run pytest -q
```

The revision appends the specific `pytest tests/unit/<new test file> -q` invocation. `pytest -q`
never collects `tests/acceptance/` (CONTRACT rule 13).

## Implementation notes

**Provisional until the lead revises the spec.**

### Locked Phase-2 values (PLAN §4.2, row "5 Hoarding")

Locked **now** and not this card's to change:

```
input_holding_loss   h_X = 0.01
alloc_eta_request    η_q = 0.7      (configuration used by WO-030/WO-031, not code here)
input_complementarity θ  = 8        (configuration used by WO-030/WO-031, not code here)
```

### 1. Input holding loss

PLAN §2.11 fixes the semantics of the **own-good** loss `h`:

```
S ← (1 − h) · S + y
```

i.e. the loss applies to carried stock *before* this period's output is added. `h_X` is the
analogous loss on **input** stocks `X_ij`. What is not fixed:

- whether `h_X` is applied at the same point of the period schedule as `h`;
- whether it acts **before or after arrivals are credited** - and with WO-022's arrival buffer in
  play, "arrivals" now means arrivals within the period, which makes the ordering a real economic
  choice rather than a formatting one.

The revision must state both. Do not infer them from the `h` convention.

Conservation (T-U1, **1e-9** per period per good): the quantity `h_X` destroys does not disappear
from the identity. It appears as an explicit loss term, exactly as `h` does.

Sign convention: `h_X ∈ [0, 1]` is a *fraction lost per period*, so `h_X = 0.0` (Phase 1) is the
identity and `h_X = 0.01` (the locked Phase-2 value) destroys 1% of carried input stock per period.

### 2. Targeted audits

PLAN §2.7.4:

```
p_i = a · (1 + κ_t · downstream_shortfall_i),  clipped into [0, 1]
```

drawn through `gosplan.rng.draw(seed_env, "audit", t, i)` as a Bernoulli - **the same purpose and
key structure as the `random` mode**, so common random numbers survive the mode switch and an
audit-mode contrast is not confounded by a different RNG stream (CONTRACT rule 9, PLAN §2.15).

- `a = information.audit_rate` is the base rate; `κ_t` scales how much visible downstream shortfall
  raises it.
- Clipping is into `[0, 1]` because `p_i` is a probability. Clipping is not an error path; it is part
  of the rule.
- `downstream_shortfall_i` reaches the planner **only** through `make_planner_view`, and only as the
  noisy, visibility-gated quantity that `PlannerView` carries (CONTRACT rule 5, T-B4). The targeted
  audit may never key on true stock.
- `targeted` is conditional on `shortfall_visibility > 0`; at `shortfall_visibility = 0.0` the branch
  reduces to `random`, exactly.

**`κ_t` is not a field of `InformationConfig` today.** The revision adds it, with a
`spec/CHANGELOG.md` entry naming version, reason and affected work orders, and classifies its arm
(INFO) per CONTRACT rule 11. **No caller may assume a value for it before then**, and this card does
not add it to `spec/spec.py` itself.

Arm bookkeeping (PLAN §4.3): `shortfall_visibility → 1` is one of the five C_OGAS moves.
`audit_rate` and `audit_noise` are C_AUDIT, dual-classified, reported separately, and **never folded
into C_OGAS**.

### 3. Soft budget - the open question, and the reason this card is MID-strong

PLAN §3 defines `soft_budget` as "bailout probability on `fill < 1`" and nothing else. CONTRACT rule
4 fixes the reward as exactly

```
production step:  −scale · c_ik
report step:       scale · (B(ρ) − 1[audited] · Pen + trade_surplus)
```

with no auxiliary term. **A bailout therefore cannot be a new reward term.** It has to move an
existing quantity. The revision must choose between the following candidates, and this card may pick
none of them on its own:

- **(a) a physical top-up** of the buyer's input stock `X_bj` when `fill < 1`. A transition rule, so
  rule-4 safe - but it changes the conservation identity of T-U1 and must appear in it explicitly as
  an injected quantity.
- **(b) a probability that the target is not ratcheted down**, or that `T_min` is relaxed. A planner
  rule, so rule-4 safe - and it may key only on `PlannerView` quantities per CONTRACT rule 5.
- **(c) a probability that an assessed penalty is waived.** Inside the existing `Pen` term, so
  rule-4 safe - but it interacts with the padding elasticity of PLAN §4.5, which is a G2 acceptance
  criterion, so its effect on that measurement has to be understood before it is chosen.

Until the revision picks one **and states the draw key** (which purpose, which indices), implementing
any of them is invention. `soft_budget = 0.0` is the Phase-1 default and must be exactly inert.

### Edge cases across all three

- All three mechanisms must short-circuit at their Phase-1 defaults with explicit early branches, so
  golden parity (T-B7, 1e-9) holds bit-for-bit and no RNG draw is consumed that was not consumed
  before.
- T-U8 must stay green: `positive_part` gives 0 for any under-report, and `audited = False` gives 0.
  A bailout implemented as a penalty waiver must not turn either of those into a non-zero.
- Measurement window is PLAN §4.4, `t ≥ 2`; a per-period loss must be well defined from `t = 0`.

Do not pick any open item. CONTRACT rule 3: choosing "the reasonable default" is a violation.

## Forbidden

- Computing, plotting or testing PLAN §4.1 rows **2, 5, 6, 7** - held out. Row 5 (hoarding) is this
  mechanism's own phenomenon and is computed for the first time in WO-030, inside the P2 acceptance
  run of WO-031 only. Note that T-B3 asserts shortage *propagation* only; the direction of hoarding
  is asserted nowhere, by design.
- Any reward term outside CONTRACT rule 4, and any bailout implemented as a reward bonus.
- Any planner function taking a `State` other than `make_planner_view` (CONTRACT rule 5); the
  targeted audit may key only on `PlannerView.downstream_shortfall`, never on true stock.
- Any rule that makes an enterprise request more inputs, or hold them longer. Hoarding must be a
  consequence of the allocation weights and of `h_X` (CONTRACT rule 7).
- Direct `numpy.random` / `jax.random` under `gosplan/env/` (CONTRACT rule 9).
- Changing the PLAN §4.2 locked values, or folding `audit_rate` / `audit_noise` into the C_OGAS
  contrast (PLAN §4.3: dual-classified, always reported separately).
- Adding `κ_t` to `spec/spec.py` from this card, or assuming a value for it.

## Contract rules that bite here

- **Rule 3 - STOP AND REPORT.** The soft-budget mechanism, the `h_X` schedule position and the value
  of `κ_t` are all undecided. Choosing "the reasonable default" for any of them is a violation; file
  an ambiguity report and end the session.
- **Rule 12 - work orders.** Read only the whitelist, write only the listed files, run the
  completion command verbatim, report in the four-part format.
- **Rule 4 - reward terms.** A bailout is not a reward term. It moves an existing quantity or it does
  not exist.
- **Rule 5 - planner blindness.** Targeted audits key on `PlannerView.downstream_shortfall` only.
- **Rule 7 - no hard-coded pathology.** Nothing here may make an enterprise request or hold more.
- **Rule 9 - RNG.** The audit draw and any bailout draw go through `rng.draw`, keyed and
  order-independent.
- **Rule 11 - parameter arms.** `κ_t` is INFO by construction; classifying it is a LEAD change with
  a CHANGELOG entry.
- **Rule 1 - frozen spec.** Only the lead edits `spec/spec.py`, and only with a CHANGELOG entry.

## Definition of done

- [ ] Every must-pass test green, including the frozen Phase-1 suite unchanged
- [ ] At the Phase-1 defaults all three mechanisms are exactly inert and every golden file passes
      unchanged
- [ ] `h_X` appears as an explicit loss term in the T-U1 conservation identity
- [ ] `uv run ruff check . && uv run ruff format --check .` clean
- [ ] `uv run python scripts/contract_guard.py --all` clean
- [ ] No file outside "Write only" created, edited or deleted
- [ ] CI `ci-ok` green on the PR
- [ ] The four-part report posted as the PR body: 1) tests passed/failed with output, 2) ambiguity
      reports filed, 3) files changed, 4) anything you were tempted to invent and did not
